### ISSUE TITLE
Can't reproduce experimental results

### ISSUE THREAD

**Issue by Alxead** [mila-iqia/atari-representation-learning#69]
I tried to run probing tasks for different Atari environments, using the following command: 

`python -m scripts.run_probe --method infonce-stdim --env-name {env_name}`

I did not change any code, just tried different game, including `PongNoFrameskip-v4`, `BowlingNoFrameskip-v4`, `BreakoutNoFrameskip-v4`, `HeroNoFrameskip-v4`. 

However, only the F1 score for `pong` matches the score reported in the paper. The F1 scores of the other three games are far worse than the score shown in the paper (for `bowling`, I got 0.22).

I check the training loss logged in wandb, it seems that training has not converged at all. See the figure below.

![training loss](https://tva1.sinaimg.cn/large/008eGmZEgy1gn16hi89crj328m0u043u.jpg)

How to get the F1 socres reported in the paper? Am I missing something?

**Comment by ankeshanand** [comment]
Hi Alex, 

That's quite strange, those loss curves don't match what we saw in our runs. There definitely seems to be a bug or a numerical error in your runs since some of the loss values are blowing up to quite high values. 

Can you confirm what pytorch version you are using? Also, I just looked up on our W&B dashboard, and this is how the loss curves look like for Bowling: 

![image](https://user-images.githubusercontent.com/2134563/105833547-5672da00-5f97-11eb-9586-7acea7ac14bd.png)

**Comment by Alxead** [comment]
My pytorch version is 1.7.0

**Comment by ankeshanand** [comment]
We used pytorch 1.1.0 for our runs back then

**Comment by ankeshanand** [comment]
So check if an older version of pytorch gives the correct results, this does look like a numerical stability bug. Also try increasing Adam's eps value. If that doesn't fix it, it's possible (though unlikely since we were quite careful) that a bug was introduced when we were cleaning up the code. I can try to debug the source of divergence then, but might only get time after the ICML deadline.

**Comment by Alxead** [comment]
Thank you, let me try it first.
